backend/database_handler/transactions_processor.py

```
@@ -874,63 +900,72 @@ def get_transactions_for_address(
+                        timestamp_appeal = :ts
+                    WHERE hash = :hash
+                      AND status IN ('ACCEPTED', 'UNDETERMINED', 'LEADER_TIMEOUT', 'VALIDATORS_TIMEOUT')
+                    """
+                ),
+                {"hash": transaction_hash, "appeal": appeal, "ts": int(time.time())},
+            )
+            if result.rowcount > 0:
+                self.session.commit()
 
     def set_transaction_timestamp_awaiting_finalization(
         self, transaction_hash: str, timestamp_awaiting_finalization: int = None
```

Ruff (0.15.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)